net/instaweb/rewriter/css_inline_filter.cc

```diff
@@ -111,11 +112,21 @@ void CssInlineFilter::StartDocumentImpl() {
 
 CssInlineFilter::~CssInlineFilter() {}
 
+void CssInlineFilter::StartElementImpl(HtmlElement* element) {
+
+  // w3c incompatible when css inlining in body with pedantic enabled
+  if (element->keyword() == HtmlName::kBody &&
+      driver()->options()->Enabled(RewriteOptions::kPedantic)) {
+    inlining_not_pedantically_spec_ = true;
```

Comment:
I am tracking whether we are in body with pedantic filter enabled. I can change this variable to in_body_ and will shift pedantic filter enable check to EndElementImpl() function, so that it will become less confusing. I will also init variable in StartDocumentImpl.